pkg/provisioner/crdprovisioner.go

```diff
@@ -212,19 +202,18 @@ func (c *CrdProvisioner) CreateVolume(
 		}
 		// if the error was caused by errors other than IsNotFound, return failure
 	} else if !apiErrors.IsNotFound(err) {
-		klog.Error("failed to get AzVolume (%s): %v", azVolumeName, err)
+		workflow.Logger().Error(err, "failed to get AzVolume CRI")
```

Comment:
We should consider converting this to an internal error and allow `Finish` to log it, e.g.:

```go
err = status.Errorf(codes.Internal, "failed to get AzVolume CRI: %v", err)
```